python_modules/dagster/dagster/serdes/serdes.py

```diff
@@ -364,17 +396,15 @@ def _unpack_value(val: Any, whitelist_map: WhitelistMap, descent_path: str) -> A
                 f"expected versions.{_path_msg(descent_path)}"
             )
 
-        klass, serializer, sig_params = whitelist_map.get_tuple_entry(klass_name)
+        klass, serializer, args_for_class = whitelist_map.get_tuple_entry(klass_name)
 
         # Target class being set to none, likely by
         if klass is None:
             return None
 
-        unpacked_val = {
-            key: _unpack_value(value, whitelist_map, f"{descent_path}.{key}")
-            for key, value in val.items()
-        }
-        return serializer.value_from_storage_dict(unpacked_val, klass, sig_params)
+        return serializer.value_from_storage_dict(
```

Comment:
A little odd that we're delegating the recursive unpack to the `NamedTupleSerializer` (which is different than how the `EnumSerializer` works).

Might be worth dropping a comment in here highlighting the reasoning?

Reply:
`EnumSerializer` / enums are leaf values so recursive unpacking isn't relevant